backend/webapp/src/main/java/org/bonitasoft/web/designer/model/asset/Asset.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (C) 2015 Bonitasoft S.A.
+ * Bonitasoft, 32 rue Gustave Eiffel - 38000 Grenoble
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2.0 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.bonitasoft.web.designer.model.asset;
+
+import static org.apache.commons.lang3.builder.ToStringStyle.SHORT_PREFIX_STYLE;
+
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonView;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.bonitasoft.web.designer.model.Identifiable;
+import org.bonitasoft.web.designer.model.JsonViewPersistence;
+import org.hibernate.validator.constraints.NotBlank;
+
+/**
+ * A web resource can be attached to a component
+ */
+public class Asset<T extends Identifiable> {
+
+    @NotBlank(message = "Asset name should not be blank")
+    @Pattern(regexp = "\\w*[.]+[a-z]{2,3}$", message = "Asset name should contains only alphanumeric characters, with no space, and the file extension in lower case")
```

Comment:
IMHO this regexp is too restrictive. We need to support things like that `jquery-1.11.3.min.js` because since we have no version attribute on Asset object, it will be written in name